docs-api/frontity-packages/collections-packages/hooks/infinite-scroll-hooks.md

```diff
@@ -0,0 +1,342 @@
+  - [Parameters](#parameters)
+  - [Return value](#return-value)
+  - [Usage](#usage)
+- [`usePostTypeInfiniteScroll`](#useposttypeinfinitescroll)
+  - [Parameters](#parameters)
+  - [Return value](#return-value)
+  - [Usage](#usage)
+- [Demo](#demo)
+
+<!-- tocstop -->
+
+## `useInfiniteScroll`
+
+This is the core hook with the basic logic to build an infinite scroll hook.
+
+It basically receives two links, `currentLink` and `nextLink`, and returns two React refs that should be attached to react elements. The hook uses `useInView` internally to track the visibility of those elements and trigger an `actions.router.set` to update the current link or an `actions.source.fetch` to fetch the next entity. You can pass options for these `useInView` hooks as well, using the `fetchInViewOptions` and the `routeInViewOptions` params.
+
+`useInfiniteScroll` also keeps a record of the fetched & ready entities in the browser history state, in order to restore the list when you go back and forward while navigating. That record is accessible from the browser history state under the `infiniteScroll.links` array.
+
+{% hint style="info" %}
+Note: the history state is also accessible from the Frontity state, in `state.router.state`.
+{% endhint %}
+
+It was designed to be used inside a `Wrapper` component that would wrap the
+entity pointed by `currentLink`.
+
+### Parameters
+
+It requires an object with the following props:
+
+| Name                     | Type                | Default | Required | Description                                                                 |
+| :----------------------- | :------------------ | :------ | :------- | :-------------------------------------------------------------------------- |
+| **`currentLink`**        | string              | -       | yes      | The current link that should be used to start the infinite scroll.          |
+| **`nextLink`**           | string              | -       | no       | The next link that should be fetched and loaded once the user scrolls down. |
+| **`fetchInViewOptions`** | IntersectionOptions | -       | no       | The intersection observer options for fetching.                             |
+| **`routeInViewOptions`** | IntersectionOptions | -       | no       | The intersection observer options for routing.                              |
+
+{% hint style="info" %}
+The IntersectionOptions type refers to the type of the the parameters received by the [`useInView` hook](https://api.frontity.org/frontity-packages/collections-packages/hooks#parameters).
+{% endhint %}
+
+### Return value
+
+| Name              | Type      | Description                                                                                      |
+| :---------------- | :-------- | :----------------------------------------------------------------------------------------------- |
+| **`supported`**   | boolean   | Boolean indicating if the Intersection Observer is supported or not by the browser.              |
```

Comment:
I would clarify here that, in the case `supported` is `false`, all the other returned properties will be `undefined`.